### DISCUSSION TITLE
RuntimeError: Failed to create a .NET runtime (coreclr) using the                 parameters {}.

### DISCUSSION THREAD

**Discussion by ReddyRomantic** [dotnet/runtime#92659] [Q&A]
Is there a way to use the .Net dlls without actually installing the .Net core Runtime or the .Net core SDK in python?

**Comment by Maximys** [comment]
I think, you could use standalone executable for that ([link](https://learn.microsoft.com/en-us/dotnet/core/deploying/single-file/overview?tabs=cli)), but I had never use it in practice, that's why I may be wrong

**Reply by ReddyRomantic** [reply]
For exe related .Net applications it is ok. But we are trying to build the DLL and use it in python inside the linux machine where .Net is no installed.. Any ideas for this?

**Reply by neon-sunset** [reply]
Rather than using https://learn.microsoft.com/en-us/dotnet/core/tutorials/netcore-hosting (which you *have* to do if you want to load .NET assemblies and call methods on them for classic .NET dlls), you can instead try https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/?tabs=net8plus%2Cwindows#build-native-libraries + https://github.com/dotnet/samples/blob/main/core/nativeaot/NativeLibrary/README.md#exporting-methods

This allows you to export C# methods and call them in the same way you would do so with C or C++ (completely standalone natively compiled binary).

**Reply by ReddyRomantic** [reply]
I am getting the following exception when I tried to apply you method.

System.BadImageFormatException: Bad IL format. The format of the file 'd:\New folder\pythondotnet\pythondotnet\NativeLibrary.dll' is invalid.

   at System.Runtime.Loader.AssemblyLoadContext.LoadFromPath(IntPtr ptrNativeAssemblyLoadContext, String ilPath, String niPath, ObjectHandleOnStack retAssembly)

   at System.Runtime.Loader.AssemblyLoadContext.LoadFromAssemblyPath(String assemblyPath)

   at System.Reflection.Assembly.LoadFrom(String assemblyFile)

   at Python.Runtime.AssemblyManager.LoadAssemblyFullPath(String name) in /tmp/build-via-sdist-zkttv147/pythonnet-3.0.2/src/runtime/AssemblyManager.cs:line 252

   at Python.Runtime.CLRModule.AddReference(String name) in /tmp/build-via-sdist-zkttv147/pythonnet-3.0.2/src/runtime/Types/ClrModule.cs:line 111

What I did?
1. I have downloaded the sample project from the microsoft sample repo.
2. Published the project as mentioned in the instruction.
3. Copied the files from the publish folder (also tried with the one inside the native folder). 
4. Pasted in my python project and added as a reference to project by using 'clr'
5. I am getting the above exception.
